### PR TITLE
Edit profile

### PR DESCRIPTION
**Pull request by jeetbhalani15**
## Added Edit Profile Functionality
- Added API calls to edit  profile details
- Edit profile image

### CODE REVIEW

File: src/Components/User-edit-modal/EditModal.jsx [hunk 1]
```diff
@@ -1,32 +1,72 @@
...
 import logo from "../../Assets/Images/logo.png"
+import { editUser } from '../../features/Auth/authSlice'
+import { useDispatch, useSelector } from 'react-redux'
+
+const EditModal = ({setShow, userImage, setUserImage}) => {
+  const dispatch = useDispatch();
+  const users = useSelector(state => state.users)
+  const authUserData = useSelector(state => state.user)
+  const [userData,setUserData] = useState({...authUserData.user})
+  console.log(userData)
```
Comment (Jay-Gadhiya) [inline src/Components/User-edit-modal/EditModal.jsx:15]: you can remove console logs

File: src/Components/User-edit-modal/EditModal.jsx [hunk 2]
```diff
@@ -1,32 +1,72 @@
...
-            <label className='p-1'>Name:</label><span className=' font-semibold'>jeeet</span>
+            <label className='p-1'>Name:</label><span className=' font-semibold'>{`${userData.firstName} ${userData.lastName}`}</span>
             </div>
             <div className='flex items-center gap-2'>
-            <label className='p-1'>username:</label><span className=' font-semibold'>jeeet13</span>
+            <label className='p-1'>username:</label><span className=' font-semibold'>{userData.username}</span>
             </div>
             <div className='flex items-center gap-2'>
-            <label className='p-1'>Website:</label><input className='p-1' type="link" placeholder='link...' />
+            <label className='p-1'>Website:</label><input onChange={(e)=>userDataChangeHandler(e)} name="website" value={userData.website}  className='p-1' type="text" placeholder='link...' />
```
Comment (Jay-Gadhiya) [inline src/Components/User-edit-modal/EditModal.jsx:64]: you can give **htmlFor** attribute for batter code readability and consistency.

File: src/features/Auth/authSlice.js [hunk 3]
```diff
@@ -63,14 +114,43 @@ const authSlice = createSlice({
...
+            state.error = action.payload
+        })
+        builder.addCase(fetchUserData.pending, (state) => {
+            state.users = []
+        })
+        builder.addCase(fetchUserData.fulfilled, (state,action) => {
+            state.users = action.payload
+        })
+        builder.addCase(fetchUserData.rejected, (state,action) => {
+            state.error = action.payload
```
Comment (Jay-Gadhiya) [inline src/features/Auth/authSlice.js:140]: I like that you put builder.addCse instead of simple case

**Review by Jay-Gadhiya — APPROVED** [review]
- code looks good 
- API and dispatch calls are written well

**Review by chanchal-panpaliya — APPROVED** [review]
code looks clean with proper naming with good UI

**Pull request merged**